coralproject/talk
# [CORL-2704] Add data cache to reduce repetitive mongo queries

## What does this PR do?

Adds a `commentCache` and `userCache` used with the `Comments.forStory` loader and the `Comment.author` resolver to up-front query a stories comments and users in consolidated queries to reduce query load on Mongo.

## These changes will impact:


## What changes to the GraphQL/Database Schema does this PR introduce?

None

## Does this PR introduce any new environment variables or feature flags? 

No

## If any indexes were added, were they added to `INDEXES.md`?

No new indexes.

## How do I test this PR?

- Load streams in standard comments mode, Q&A, and R&R mode.
- Ensure stream loads and updates properly.
 
## How do we deploy this PR?

No special considerations for deployment.
